pubpub/pubpub
# feature: Table breakout (and smaller font)

Closes #754

This PR adds two new features to tables:

- A "smaller font" option that makes table text 20% smaller
- A "breakout" menu that lets tables break out of the Pub body the same way figures can, at some set fractions of viewport sizes.

These appear as two new menu items in the hovering table menu:

![image](https://user-images.githubusercontent.com/2208769/214113443-7c37d04c-a884-435e-8bbd-d1176be0b84f.png)

Table breakouts at 50% and 75% look kind of funny on smaller viewports (and they revert back to the Pub column on mobile layouts). But we talked it over and decided we're going to introduce this to the editor as-is and see what people say.

This PR is quite dense with changes — I had to do some refactoring to the formatting bar state management to permit the new nested breakout menu, but I think it makes the code easier to understand.

_Test plan:_
- Create a table in a Pub
- Make sure you can toggle the "smaller font" option
- Make sure you can set the table breakout value
- Also just generally test the formatting bar, since I did a bit of refactoring there.
